### PR TITLE
added modules vector_ques and hash_map_ques in Feature/assignment 10

### PR DESCRIPTION
**Pull request by ankit-mogha**
What does this change do?
1. hash_map_ques module has hash_map.rs file that has a function sum_conditionals function matches a particular pattern in Key of hashmap.

2. vector_ques has the following modules:- 
   i. add_duplicates.rs:- have a duplicate function that duplicates the elements of the list. 
   ii. first_even_number.rs:- have a  first_even function that checks the first even number in the list.
   iii. list_palindrome.rs:- have an is_palindrome function checks if the list is palindrome or not.
   iv. remove_duplicate.rs:- have a compress function that removes duplicates elements from the list.
   v. remove_nth_element:- have a drop function that finds the given number in the list and removes it.
   vi. reverse_list.rs:- have a reverse function that reverses the list.

Any additional information for the reviewer to start
NA

How should this be manually tested?
We need window OS in which Rust is installed. then execute this rust program

Which Trello task and sub-task does this change relate to?

Are there any changes pending?
No

Does any team have to be notified of changes in this feature?
Yes

Definition of Done:
- [x] Is there >90% unit test code coverage?
if not please mention the Code Coverage of code.
- [x] Does this PR add new dependencies? If so, please list out the same.
      -  log
- [ ]  Will this feature require a new piece of infrastructure to be implemented?
- [x] Is there appropriate logging included?
- [x] Does the project compile ok?
- [x] Have Clippy violations been fixed?
- [x] Have the Code is properly formatted?

### CODE REVIEW

File: assignments/assignment_ten/src/vector_ques/reverse_list.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    let mut res: Vec<i32> = Vec::new();
+    let mut len = vec.len();
+    while len > 0 {
+        let last = vec.last();
+        match last {
+            Some(ele) => {
+                res.push(*ele);
+                vec.pop();
+            }
+            None => println!("no"),
```
Comment (aman09v) [inline assignments/assignment_ten/src/vector_ques/reverse_list.rs:20]: do not use print statements

**Review by aman09v — CHANGES_REQUESTED** [review]

File: assignments/assignment_ten/src/vector_ques/reverse_list.rs (hunk 1 above)
Comment (ankit-mogha) [reply assignments/assignment_ten/src/vector_ques/reverse_list.rs]: updated the file

**Review by aman09v — APPROVED** [review]

**Review by sonukumar-a11 — APPROVED** [review]
Use a single function for particular module test

**Review by Dheerajjnv — APPROVED** [review]

File: assignments/assignment_ten/src/vector_ques/add_duplicates.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/// #Arguments
+///
+/// vec : its a list of elements vector type.
+///
+/// #Return
+///
+/// Returns list of vector type having duplicate values.
+pub fn duplicate(vec: Vec<i32>) -> Vec<i32> {
+    let mut res: Vec<i32> = Vec::new();
+    println!("{}", vec.len());
```
Comment (NitinSaxenait) [inline assignments/assignment_ten/src/vector_ques/add_duplicates.rs:12]: println ?? not needed

File: assignments/assignment_ten/src/vector_ques/add_duplicates.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    for index in vec {
+        res.push(index);
+        res.push(index);
+    }
+    log::info!("List : {:?}", res);
```
Comment (NitinSaxenait) [inline assignments/assignment_ten/src/vector_ques/add_duplicates.rs:17]: wrong use of info.

File: assignments/assignment_ten/src/vector_ques/remove_duplicate.rs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        match top {
+            Some(e) => {
+                if value != *e {
+                    res.push(value);
+                }
+            }
+            None => res.push(value),
+        }
+    }
+    log::info!("List : {:?}", res);
```
Comment (NitinSaxenait) [inline assignments/assignment_ten/src/vector_ques/remove_duplicate.rs:23]: info use!! not correct.

**Review by NitinSaxenait — APPROVED** [review]

**Comment by NitinSaxenait** [comment]
Kya kr rhe ho ankit bhai. Damad bna lia kya knoldus ne

**Pull request merged**